.travis.yml

```diff
@@ -7,6 +7,15 @@ addons:
 
 python:
   - "2.7"
+  - "3.7"
+
+matrix:
+  allow_failures:
+    - python: "3.7"
+
+before_script:
+  - pip install flake8
+  - flake8 . --count --select=E9,F63,F72,F82 --show-source --statistics
```

Comment:
@cclauss I'm not an expert on travis-cli. What is the effect of adding flake8 checks before the tests? Will the build fail if the entire framework doesn't pass flake8?

Reply:
As well, what's the justification behind the selected checks?

Reply:
There are two advantages to putting the flake8 checks in the __before_script__:
1. flake8 processes to codebase really quickly and these are fast-fail tests.  We want to give committers rapid feedback if we spot a typo or syntax error in their submission.
2. Travis CI hides the output of successful __before_script__ commands which reduces visual clutter so that committers can more rapidly find failing tests.  This is especially useful __flake8 --exit-zero__ runs but is also useful elsewhere.

On the test selection, I have no interest in "_style violations_" (the majority of flake8 error codes that [__python/black__](https://github.com/python/black can autocorrect).  Instead these test focus on runtime safety and correctness:
* E9 tests are about Python syntax errors usually raise because flake8 can not build an Abstract Syntax Tree (AST).  Often these issues are a sign of unused code or code that has not been ported to Python 3.  These would be compile-time errors in a compiled language but in a dynamic language like Python they result in the script halting/crashing on the user.
* F63 tests are usually about the confusion between identity and equality in Python.  Use ==/!= to compare str, bytes, and int literals is the classic case.  These are areas where __a == b__ is True but __a is b__ is False (or vice versa).
* F72 tests syntax errors in type hints
* F82 tests are almost always _undefined names_ which are usually a sign of a typo, missing imports, or code that has not been ported to Python 3.  These also would be compile-time errors in a compiled language but in Python a __NameError__ and the script halting/crashing on the user.